src/libraries/Common/tests/WasmTestRunner/WasmTestRunner.cs

```diff
@@ -30,41 +34,105 @@ public static async Task<int> Main(string[] args)
             switch (option)
             {
                 case "-notrait":
-                    excludedTraits.Add (args[i + 1]);
+                    excludedTraits.Add(args[i + 1]);
                     i++;
                     break;
                 case "-trait":
-                    includedTraits.Add (args[i + 1]);
+                    includedTraits.Add(args[i + 1]);
                     i++;
                     break;
                 case "-namespace":
-                    includedNamespaces.Add (args[i + 1]);
+                    includedNamespaces.Add(args[i + 1]);
                     i++;
                     break;
                 case "-class":
-                    includedClasses.Add (args[i + 1]);
+                    includedClasses.Add(args[i + 1]);
                     i++;
                     break;
                 case "-method":
-                    includedMethods.Add (args[i + 1]);
+                    includedMethods.Add(args[i + 1]);
                     i++;
                     break;
+                case "-mt":
+                    break;
                 default:
                     throw new ArgumentException($"Invalid argument '{option}'.");
             }
         }
 
-        var runner = new SimpleWasmTestRunner()
+        WasmApplicationEntryPointBase? runner = null;
+        if (args.Contains("-mt"))
+        {
+            Console.WriteLine("Using MultiThreadedTestRunner");
+            runner = new MultiThreadedTestRunner()
+            {
+                TestAssembly = testAssembly,
+                ExcludedTraits = excludedTraits,
+                IncludedTraits = includedTraits,
+                IncludedNamespaces = includedNamespaces,
+                IncludedClasses = includedClasses,
+                IncludedMethods = includedMethods
+            };
+        }
+        else
         {
-            TestAssembly = testAssembly,
-            ExcludedTraits = excludedTraits,
-            IncludedTraits = includedTraits,
-            IncludedNamespaces = includedNamespaces,
-            IncludedClasses = includedClasses,
-            IncludedMethods = includedMethods
-        };
+            Console.WriteLine("Using SimpleWasmTestRunner");
+            runner = new SimpleWasmTestRunner()
+            {
+                TestAssembly = testAssembly,
+                ExcludedTraits = excludedTraits,
+                IncludedTraits = includedTraits,
+                IncludedNamespaces = includedNamespaces,
+                IncludedClasses = includedClasses,
+                IncludedMethods = includedMethods
+            };
+        }
 
         await Task.Yield();
```

Comment:
This yield doesn't work on wasi. I'm fixing it here https://github.com/dotnet/runtime/pull/94599